### PR TITLE
fix(dropdown): add horizontal repositioning when space is not available

### PR DESCRIPTION
**Pull request by QuintonJason**
## Description

- [x] Rails dropdown update horizontal repositioning so that modal doesn't hang off the screen
- [x] React dropdownupdate horizontal repositioning so that modal doesn't hang off the screen 

#### Rails
Kapture 2023-10-18 at 16 24 48

#### React
Kapture 2023-10-18 at 17 04 01

## Testing in `sage-lib`

- React testing steps - https://www.loom.com/share/322f3bc3056b41e8949186b454ffd2a1?sid=690d3b81-b2b1-4f22-8061-263fafcf6a26
- Rails testing steps - https://www.loom.com/share/a952416299894a39981280bcb01a5453?sid=2edb2e7e-5d0b-4e71-8d0c-9df1541833de

## Testing in `kajabi-products`

1. (**HIGH**) Updates the dropdown positioning function include horizontal repositioning.

## Related

Closes [DSS-477](https://kajabi.atlassian.net/browse/DSS-477)

[DSS-477]: https://kajabi.atlassian.net/browse/DSS-477?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/sage-system/lib/dropdown.js [hunk 1]
```diff
@@ -190,49 +190,82 @@ Sage.dropdown = (function () {
...
+    let directionY = null;
+  
     // Elements
     const button = el;
-    const panel = el.lastElementChild
+    const panel = el.lastElementChild;
     const win = panel.ownerDocument.defaultView;
     const docEl = window.document.documentElement;
-
+
```
Comment (ju-Skinner) [inline packages/sage-system/lib/dropdown.js:201]: looks like you have extra whitespace here.

File: packages/sage-system/lib/dropdown.js [hunk 2]
```diff
@@ -190,49 +190,82 @@ Sage.dropdown = (function () {
...
     panel.style.top = ''; // resets the style
-
+    panel.style.left = ''; // resets the style
+    panel.style.right = ''; // resets the style
+  
     // Dimensions
     const buttonDimensions = button.getBoundingClientRect();
     const panelDimensions = panel.getBoundingClientRect();
-
+
```
Comment (ju-Skinner) [inline packages/sage-system/lib/dropdown.js:209]: extra whitespace here as well.

File: packages/sage-system/lib/dropdown.js [hunk 3]
```diff
@@ -190,49 +190,82 @@ Sage.dropdown = (function () {
...
+  
     const offset = {
       top: panelDimensions.top + win.pageYOffset,
       left: panelDimensions.left + win.pageXOffset,
-      bottom: (panelDimensions.top + win.pageYOffset)
+      bottom: (panelDimensions.top + win.pageYOffset),
+      right: (panelDimensions.left + win.pageXOffset),
     };
-
+
```
Comment (ju-Skinner) [inline packages/sage-system/lib/dropdown.js:228]: extra whitespace here as well.

File: packages/sage-system/lib/dropdown.js [hunk 4]
```diff
@@ -190,49 +190,82 @@ Sage.dropdown = (function () {
...
     if (!enoughSpaceBelow && enoughSpaceAbove) {
-      direction = 'above';
+      directionY = 'above';
     } else if (!enoughSpaceAbove && enoughSpaceBelow) {
-      direction = 'below';
+      directionY = 'below';
     }
-
-    if ( direction == 'above') {
+
```
Comment (ju-Skinner) [inline packages/sage-system/lib/dropdown.js:242]: extra whitespace here as well.

**Review by ju-Skinner — CHANGES_REQUESTED** [review]
Just fix the whitespace issues. Other than that things are good.

File: packages/sage-system/lib/dropdown.js [hunk 5]
```diff
@@ -190,49 +190,82 @@ Sage.dropdown = (function () {
...
+    if (directionX === 'left') {
+      panel.style.left = 'inherit';
+      panel.style.right = 0;
+    } else if (directionX === 'right') {
+      panel.style.left = 0;
+      panel.style.right = 'inherit';
     }
+
+    console.log('directionX', directionX);
+    console.log('directionY', directionY);
```
Comment (pixelflips) [inline packages/sage-system/lib/dropdown.js:266]: Should these be removed?

**Review by pixelflips — APPROVED** [review]
A couple of `console.log` and the whitespace mentioned by Julian, otherwise great work!

**Review by pixelflips — APPROVED** [review]

**Review by ju-Skinner — APPROVED** [review]

**Review by anechol — APPROVED** [review]
🚀

**Pull request merged**